[CI] Many tests failing when testing with fips

This started occurring with #109899 

They are failing with:

```

Caused by: java.lang.IllegalStateException: jar hell! |  
-- | --
  | class: org.bouncycastle.LICENSE |  
  | jar1: /opt/buildkite-agent/.gradle/caches/modules-2/files-2.1/org.bouncycastle/bcprov-jdk18on/1.78.1/39e9e45359e20998eb79c1828751f94a818d25f8/bcprov-jdk18on-1.78.1.jar |  
  | jar2: /opt/buildkite-agent/.gradle/caches/modules-2/files-2.1/org.bouncycastle/bc-fips/1.0.2.4/9008d04fc13da6455e6a792935b93b629757335d/bc-fips-1.0.2.4.jar
```

**Build Scans:**
- [elasticsearch-periodic #3265 / openjdk17_checkpart4_java-fips-matrix](https://gradle-enterprise.elastic.co/s/vcftxhhnqladw)
- [elasticsearch-pull-request #24283 / part-4-fips](https://gradle-enterprise.elastic.co/s/gi3kirqtrgdri)
- [elasticsearch-pull-request #24279 / part-4-fips](https://gradle-enterprise.elastic.co/s/po7p4enaatonm)
- [elasticsearch-pull-request #24276 / part-4-fips](https://gradle-enterprise.elastic.co/s/yzjiwyuimar7u)
- [elasticsearch-pull-request #24269 / part-4-fips](https://gradle-enterprise.elastic.co/s/7yvvrmxthjrvq)
- [elasticsearch-pull-request #24268 / part-4-fips](https://gradle-enterprise.elastic.co/s/25ynh7jv2dihk)
- [elasticsearch-periodic #3262 / openjdk17_checkpart4_java-fips-matrix](https://gradle-enterprise.elastic.co/s/5poffknbw3pgu)
- [elasticsearch-pull-request #24256 / part-4-fips](https://gradle-enterprise.elastic.co/s/ewntvjww2cbny)
- [elasticsearch-pull-request #24213 / part-4-fips](https://gradle-enterprise.elastic.co/s/66oqsdegdwnl6)
- [elasticsearch-pull-request #24213 / part-4-fips](https://gradle-enterprise.elastic.co/s/bzf7nm3iylhc2)

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('classMethod'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.security.authc.file.tool.UsersToolTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.RuntimeException: found jar hell in test classpath
```

**Issue Reasons:**
- [main] 17 consecutive failures in test classMethod (100.0% fail rate)
- [main] 9 consecutive failures in step openjdk17_checkpart4_java-fips-matrix (100.0% fail rate)
- [main] 8 consecutive failures in step part-4-fips (100.0% fail rate)
- [main] 9 consecutive failures in pipeline elasticsearch-periodic (100.0% fail rate)
- [main] 7 consecutive failures in pipeline elasticsearch-pull-request (100.0% fail rate)
- [main] 17 failures in test classMethod (100.0% fail rate)
- [main] 9 failures in step openjdk17_checkpart4_java-fips-matrix (100.0% fail rate)
- [main] 8 failures in step part-4-fips (100.0% fail rate)
- [main] 9 failures in pipeline elasticsearch-periodic (100.0% fail rate)
- [main] 7 failures in pipeline elasticsearch-pull-request (100.0% fail rate)

--- a/x-pack/plugin/security/build.gradle
+++ b/x-pack/plugin/security/build.gradle
@@ -151,11 +151,6 @@ dependencies {
   testImplementation('org.apache.directory.server:ldap-client-test:2.0.0-M24')
   testImplementation('org.apache.directory.server:apacheds-interceptor-kerberos:2.0.0-M24')
   testImplementation('org.apache.directory.mavibot:mavibot:1.0.0-M8')
-
-  // netty self signed certificate dependency
-  testImplementation('org.bouncycastle:bcprov-jdk18on:1.78.1')
-  testImplementation ('org.bouncycastle:bcutil-jdk18on:1.78.1')
-  testImplementation('org.bouncycastle:bcpkix-jdk18on:1.78.1')
 }
 
 tasks.named("test").configure {

--- a/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4TransportCloseNotifyIT.java
+++ b/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4TransportCloseNotifyIT.java
@@ -25,18 +25,15 @@ import io.netty.handler.codec.http.HttpVersion;
 import io.netty.handler.ssl.SslContextBuilder;
 import io.netty.handler.ssl.SslHandler;
 import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
-import io.netty.handler.ssl.util.SelfSignedCertificate;
 
 import org.elasticsearch.action.admin.cluster.state.ClusterStateAction;
 import org.elasticsearch.action.support.CancellableActionTestPlugin;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.util.CollectionUtils;
-import org.elasticsearch.core.SuppressForbidden;
 import org.elasticsearch.http.HttpServerTransport;
 import org.elasticsearch.plugins.Plugin;
 import org.elasticsearch.test.ESIntegTestCase.ClusterScope;
 import org.elasticsearch.test.ESIntegTestCase.Scope;
-import org.elasticsearch.test.ESTestCase;
 import org.elasticsearch.test.SecurityIntegTestCase;
 
 import java.util.Collection;
@@ -45,12 +42,11 @@ import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.Consumer;
 
+import static org.elasticsearch.test.SecuritySettingsSource.addSSLSettingsForNodePEMFiles;
 import static org.elasticsearch.test.TaskAssertions.assertAllTasksHaveFinished;
 import static org.elasticsearch.test.rest.ESRestTestCase.basicAuthHeaderValue;
 
 @ClusterScope(numDataNodes = 0, scope = Scope.TEST)
-@ESTestCase.WithoutSecurityManager
-@SuppressForbidden(reason = "requires java.io.File for netty self-signed certificate")
 public class SecurityNetty4TransportCloseNotifyIT extends SecurityIntegTestCase {
 
     @Override
@@ -60,17 +56,9 @@ public class SecurityNetty4TransportCloseNotifyIT extends SecurityIntegTestCase 
 
     @Override
     protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
-        try {
-            var ssc = new SelfSignedCertificate();
-            return Settings.builder()
-                .put(super.nodeSettings(nodeOrdinal, otherSettings))
-                .put("xpack.security.http.ssl.enabled", true)
-                .put("xpack.security.http.ssl.key", ssc.privateKey().getPath())
-                .put("xpack.security.http.ssl.certificate", ssc.certificate().getPath())
-                .build();
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
+        final Settings.Builder builder = Settings.builder().put(super.nodeSettings(nodeOrdinal, otherSettings));
+        addSSLSettingsForNodePEMFiles(builder, "xpack.security.http.", randomBoolean());
+        return builder.put("xpack.security.http.ssl.enabled", true).build();
     }
 
     @Override

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4HttpServerTransportCloseNotifyTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/netty4/SecurityNetty4HttpServerTransportCloseNotifyTests.java
@@ -25,11 +25,9 @@ import io.netty.handler.codec.http.HttpVersion;
 import io.netty.handler.ssl.SslContextBuilder;
 import io.netty.handler.ssl.SslHandler;
 import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
-import io.netty.handler.ssl.util.SelfSignedCertificate;
 import io.netty.util.concurrent.Future;
 
 import org.elasticsearch.common.network.NetworkService;
-import org.elasticsearch.common.settings.MockSecureSettings;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.util.concurrent.ThreadContext;
 import org.elasticsearch.core.SuppressForbidden;
@@ -62,6 +60,8 @@ import java.util.function.Consumer;
 
 import javax.net.ssl.SSLException;
 
+import static org.elasticsearch.test.SecuritySettingsSource.addSSLSettingsForNodePEMFiles;
+
 @ESTestCase.WithoutSecurityManager
 @SuppressForbidden(reason = "requires java.io.File for netty self-signed certificate")
 public class SecurityNetty4HttpServerTransportCloseNotifyTests extends AbstractHttpServerTransportTestCase {
@@ -93,17 +93,12 @@ public class SecurityNetty4HttpServerTransportCloseNotifyTests extends AbstractH
      * The server will not reply to request automatically, to send response poll the queue.
      */
     private HttpServer setupHttpServer(String tlsProtocols) throws CertificateException {
-        var ssc = new SelfSignedCertificate();
         var threadPool = new TestThreadPool("tls-close-notify");
         var dispatcher = new QueuedDispatcher();
-        var secureSettings = new MockSecureSettings();
-        secureSettings.setString("xpack.security.http.ssl.secure_key_passphrase", "testnode");
-        var settings = Settings.builder()
-            .put("xpack.security.http.ssl.enabled", true)
-            .put("xpack.security.http.ssl.key", ssc.privateKey().getPath())
-            .put("xpack.security.http.ssl.certificate", ssc.certificate().getPath())
+        final Settings.Builder builder = Settings.builder();
+        addSSLSettingsForNodePEMFiles(builder, "xpack.security.http.", randomBoolean());
+        var settings = builder.put("xpack.security.http.ssl.enabled", true)
             .put("path.home", createTempDir())
-            .setSecureSettings(secureSettings)
             .put("xpack.security.http.ssl.supported_protocols", tlsProtocols)
             .build();
         var env = TestEnvironment.newEnvironment(settings);

